### PR TITLE
Refactored Search Movie Feature

### PR DESCRIPTION
**Pull request by smabhena**
The rest of the features will be refactored in coming pull requests

### CODE REVIEW

File: Mooov/Mooov/Search Movie/SearchResultsViewController.swift [hunk 1]
```diff
@@ -14,61 +14,35 @@ class SearchResultsViewController: UIViewController {
     private var searchText: String = ""
     private var results: SearchResults?
     
+    private lazy var viewModel = SearchMovieViewModel(repository: SearchMovieRepository(), delegate: self)
```
Comment (bgouws) [inline Mooov/Mooov/Search Movie/SearchResultsViewController.swift:17]: Just a small code style thing 😄 

But you can put each parameter on its own line.

```
private lazy var viewModel = SearchMovieViewModel(repository: SearchMovieRepository(),
                                                  delegate: self)
```

File: Mooov/Mooov/Search Movie/SearchResultsViewController.swift [hunk 2]
```diff
@@ -14,61 +14,35 @@ class SearchResultsViewController: UIViewController {
     private var searchText: String = ""
     private var results: SearchResults?
```
Comment (bgouws) [inline Mooov/Mooov/Search Movie/SearchResultsViewController.swift:15]: Your ViewController doesn't need to know about `SearchResults ` 😄

File: Mooov/Mooov/Search Movie/SearchResultsViewController.swift [hunk 3]
```diff
@@ -14,61 +14,35 @@ class SearchResultsViewController: UIViewController {
...
-        }
-        return 0
+        return viewModel.searchCount
     }
     
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
         let cell = tableView.dequeueReusableCell(withIdentifier: "cell", for: indexPath)
         
-        guard let movieTitle = self.results?.search[indexPath.row].title else {
+        guard let movieTitle = viewModel.results?.search[indexPath.row].title else {
```
Comment (bgouws) [inline Mooov/Mooov/Search Movie/SearchResultsViewController.swift:45]: Why not have a `func` in your `ViewModel` that returns the movieTitle ready to use?

File: Mooov/Mooov/Search Movie/SearchResultsViewController.swift [hunk 4]
```diff
@@ -84,11 +58,21 @@ extension SearchResultsViewController: UITableViewDelegate, UITableViewDataSourc
     override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
         if let movieInfoPage = segue.destination as? MovieInfoViewController {
                 guard let index = tableView.indexPathForSelectedRow?.row else { return }
-                guard let movie = results?.search[index] else { return }
+                guard let movie = viewModel.results?.search[index] else { return }
```
Comment (bgouws) [inline Mooov/Mooov/Search Movie/SearchResultsViewController.swift:61]: You can reuse that `func` I mentioned in the previous comment.

File: Mooov/Mooov/Search Movie/SearchResultsViewController.swift [hunk 5]
```diff
@@ -84,11 +58,21 @@ extension SearchResultsViewController: UITableViewDelegate, UITableViewDataSourc
...
     }
 }
+
+extension SearchResultsViewController: ViewModelDelegate {
+    func reloadView() {
+        tableView.reloadData()
+    }
+    
+    func show(error: String) {
+        displayAlert(title: "Error", message: error, buttonTitle: "Ok")
```
Comment (bgouws) [inline Mooov/Mooov/Search Movie/SearchResultsViewController.swift:76]: Each parameter on its own line 😄

File: Mooov/Mooov/Search Movie/Repository/SearchMovieRepository.swift [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+        var request = URLRequest(url: endpoint.url)
+        
+        request.httpMethod = "\(method)"
+        request.setValue(Constants.apiHost, forHTTPHeaderField: "x-rapidapi-host")
+        request.setValue(Constants.apiKey, forHTTPHeaderField: "x-rapidapi-key")
+        request.allHTTPHeaderFields = ["Content-Type": "application/json"]
+        call(with: request, completion: completion)
+    }
+    
+    private func call<T: Codable>(with request: URLRequest, completion: @escaping((Result<T, APIError>) -> Void)) {
```
Comment (simekanidvt) [inline Mooov/Mooov/Search Movie/Repository/SearchMovieRepository.swift:38]: You could make use of typealias to make this look cleaner, for this -> "(Result<T, APIError>) -> Void)"

**Review by simekanidvt — CHANGES_REQUESTED** [review]

**Review by kirsty14 — DISMISSED** [review]
Looking a lot cleaner. Nice work!

**Review by kirsty14 — APPROVED** [review]

**Review by simekanidvt — APPROVED** [review]

**Review by bgouws — APPROVED** [review]
Just those naming enhancements but looks really good 👍

File: Mooov/Mooov/Search Movie/SearchMovieViewModel.swift [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    
+    var searchCount: Int {
+        return self.movieResults?.search.count ?? 0
+    }
+    
+    var results: SearchResults? {
+        return self.movieResults
+    }
+    
+    func getMovieTitle(_ atIndex: Int) -> String? {
```
[the remark this answers is not in the document]
Comment (smabhena) [reply Mooov/Mooov/Search Movie/SearchMovieViewModel.swift:34]: Noted, will fix it in my next PR 👍

**Pull request merged**